rumqttc/src/eventloop.rs

```diff
@@ -127,14 +126,13 @@ impl EventLoop {
     pub fn clean(&mut self) {
         self.network = None;
         self.keepalive_timeout = None;
-        let mut pending = self.state.clean();
+        self.pending.extend(self.state.clean());
```

Comment:
calling `.extend()` will imply that previous elements present in pending will persist across calls to `clean()` right? is this the expected behavior desired?

asking because previously we were just updating `self.pending` ( thus, discarding the elements stored in it previously )

Reply:
> calling .extend() will imply that previous elements present in pending will persist across calls to clean() right?
Yes, we are consuming the queue of pending in fifo order, so this is taken care of by `EventLoop::poll` on system restart, the trouble why this edit had to be made is, consider the code:
```
use tokio::{task, time};

use rumqttc::{self, AsyncClient, Event, MqttOptions, Outgoing, QoS};
use std::error::Error;
use std::time::Duration;

#[tokio::main(worker_threads = 1)]
async fn main() -> Result<(), Box<dyn Error>> {
    pretty_env_logger::init();
    // color_backtrace::install();

    let mut mqttoptions = MqttOptions::new("test-1", "localhost", 1883);
    mqttoptions.set_keep_alive(Duration::from_secs(5));

    let (client, mut eventloop) = AsyncClient::new(mqttoptions, 10);
    task::spawn(async move {
        client
            .subscribe("hello/world", QoS::AtMostOnce)
            .await
            .unwrap();

        for i in 1..=10 {
            client
                .publish("hello/world", QoS::AtLeastOnce, false, vec![1; i])
                .await
                .unwrap();

            time::sleep(Duration::from_secs(1)).await;
        }
    });

    loop {
        let event = eventloop.poll().await;
        match &event {
            Ok(Event::Outgoing(Outgoing::Publish(i))) if i > &5 => {
                eventloop.clean();
                dbg!(&eventloop.pending);
            }
            Ok(v) => {
                println!("Event = {v:?}");
            }
            Err(e) => {
                println!("Error = {e:?}");
                return Ok(());
            }
        }
    }
}
```

Run the code on previous and current rumqttc to understand the effect of this change